mslice/models/slice/mantid_slice_algorithm.py

```diff
@@ -65,6 +65,19 @@ def compute_chi_magnetic(self, chi):
         chi_magnetic = chi / 291
         return chi_magnetic
 
+    def compute_d2sigma(self, scattering_data, workspace, y_axis):
+        Ei = self._workspace_provider.get_EFixed(self._workspace_provider.get_workspace_handle(workspace))
```

Comment:
This feels messy and we're also doing similar things with the temperature so it would be good to clean this up. Maybe we need some centralized functions?